### ISSUE TITLE
[Feature Request] Have the ComfyUI API return with filename of the generated files

### ISSUE THREAD

**Issue by cohan8999** [comfyanonymous/ComfyUI#1083]
Might be this is already possible? I was not able to find any documentation about it. 

How about implementing a node for outputs to the API, allowing the users to customize the outputs by forwarding strings like prompts and such to the node?

**Comment by fuami** [comment]
The API currently accepts prompts via post,
https://github.com/comfyanonymous/ComfyUI/blob/master/script_examples/basic_api_example.py

And can be monitored for status, and output via WebSockets,
https://github.com/comfyanonymous/ComfyUI/blob/master/script_examples/websockets_api_example.py

This includes getting the filenames, since the prompt contains the entire workflow you can fully customize the entire thing without limit; including submitting varied workflows, adjusting prompts, setting seed, and far, far more.

**Comment by cdjhz** [comment]
Got HTTP Error 400: Bad Request. Is the requested data properly set?

**Comment by p0mad** [comment]
@fuami @cohan8999 @dotgrid

Is there a way to run the same API for multiple queue (batch counts)?
I have a sequence image loader and also sequence image save. So now i want to run the whole folder with a single API call!

any idea would be much appritiated!
Thanks

**Comment by EmmanuelMr18** [comment]
> @fuami @cohan8999 @dotgrid
> 
> Is there a way to run the same API for multiple queue (batch counts)? I have a sequence image loader and also sequence image save. So now i want to run the whole folder with a single API call!
> 
> any idea would be much appritiated! Thanks
> 
> This is the workflow:
> 
> { "11": { "inputs": { "transparency": "false", "images": [ "14", 0 ] }, "class_type": "Image Rembg (Remove Background)" }, "14": { "inputs": { "mode": "incremental_image", "index": 0, "label": "00001.png", "path": "/home/jovyan/workspace/Control/input/orig", "pattern": "*.png", "allow_RGBA_output": "false" }, "class_type": "Load Image Batch" }, "15": { "inputs": { "output_path": "/home/jovyan/workspace/Control/input/rbg", "filename_prefix": "", "filename_delimiter": "", "filename_number_padding": 3, "filename_number_start": "false", "extension": "png", "quality": 100, "lossless_webp": "false", "overwrite_mode": "false", "show_history": "false", "show_history_by_prefix": "true", "embed_workflow": "true", "show_previews": "false", "images": [ "11", 0 ] }, "class_type": "Image Save" } }

Yes, the way that comfyUI do that is sending n request to `/prompt` endpoint.
What does that mean?
You can't do it with a single API call.
In your python script you should do it by hand:
1. get all the images of your folder 
2. With a for, you could iterate in each file and send the POST request to `/prompt`

**Comment by EmmanuelMr18** [comment]
> Got HTTP Error 400: Bad Request. Is the request data properly set?

Could be that you have a typo or other thing. I had the same problem and the way that I found to fix is the following:
1. Open the terminal where you started comfyUI
2. Send your POST request
3. Verify what is printed in the first terminal
![image](https://github.com/comfyanonymous/ComfyUI/assets/31712515/b172ce26-0b61-4d22-8db5-c4ac92f7fa57)

In my left terminal I have the request and in the right comfyUI
In the print I found that I had a typo, "ealisticVision" is not the model, I just change it to the correct one and It worked (realisticVision).

I hope that this is useful to you.

**Comment by min-star** [comment]
> Got HTTP Error 400: Bad Request. Is the request data properly set?

me too!

**Comment by huangbaichao** [comment]
> @fuami @cohan8999 @dotgrid
> 
> Is there a way to run the same API for multiple queue (batch counts)? I have a sequence image loader and also sequence image save. So now i want to run the whole folder with a single API call!
> 
> any idea would be much appritiated! Thanks
> 
> This is the workflow:
> 
> { "11": { "inputs": { "transparency": "false", "images": [ "14", 0 ] }, "class_type": "Image Rembg (Remove Background)" }, "14": { "inputs": { "mode": "incremental_image", "index": 0, "label": "00001.png", "path": "/home/jovyan/workspace/Control/input/orig", "pattern": "*.png", "allow_RGBA_output": "false" }, "class_type": "Load Image Batch" }, "15": { "inputs": { "output_path": "/home/jovyan/workspace/Control/input/rbg", "filename_prefix": "", "filename_delimiter": "", "filename_number_padding": 3, "filename_number_start": "false", "extension": "png", "quality": 100, "lossless_webp": "false", "overwrite_mode": "false", "show_history": "false", "show_history_by_prefix": "true", "embed_workflow": "true", "show_previews": "false", "images": [ "11", 0 ] }, "class_type": "Image Save" } }

After you got json from save API, you should put it as the value of prompt key, then with http post request.
<img width="752" alt="image" src="https://github.com/comfyanonymous/ComfyUI/assets/9191067/2d169b75-337c-49f4-82c7-6f66df7f4102">

**Comment by peterpernhofer** [comment]
> > @fuami @cohan8999 @dotgrid
> > Is there a way to run the same API for multiple queue (batch counts)? I have a sequence image loader and also sequence image save. So now i want to run the whole folder with a single API call!
> > any idea would be much appritiated! Thanks
> > This is the workflow:
> > { "11": { "inputs": { "transparency": "false", "images": [ "14", 0 ] }, "class_type": "Image Rembg (Remove Background)" }, "14": { "inputs": { "mode": "incremental_image", "index": 0, "label": "00001.png", "path": "/home/jovyan/workspace/Control/input/orig", "pattern": "*.png", "allow_RGBA_output": "false" }, "class_type": "Load Image Batch" }, "15": { "inputs": { "output_path": "/home/jovyan/workspace/Control/input/rbg", "filename_prefix": "", "filename_delimiter": "", "filename_number_padding": 3, "filename_number_start": "false", "extension": "png", "quality": 100, "lossless_webp": "false", "overwrite_mode": "false", "show_history": "false", "show_history_by_prefix": "true", "embed_workflow": "true", "show_previews": "false", "images": [ "11", 0 ] }, "class_type": "Image Save" } }
> 
> After you got json from save API, you should put it as the value of prompt key, then with http post request. <img alt="image" width="752" src="https://private-user-images.githubusercontent.com/9191067/280425115-2d169b75-337c-49f4-82c7-6f66df7f4102.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA3NzUyMTEsIm5iZiI6MTcxMDc3NDkxMSwicGF0aCI6Ii85MTkxMDY3LzI4MDQyNTExNS0yZDE2OWI3NS0zMzdjLTQ5ZjQtODJjNy02ZjY2ZGY3ZjQxMDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMThUMTUxNTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzdhNTY2NTYyYzNjZTEzOWE2ZjU2ZGFkMmRjZWExNjViYTMwZjg5YmIxMDYxYTNlZTRmZDQyMTEyM2JkYmU1NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.-jwUeSCAYrS8M7WFSxgqjk4-_wci-O8ngg3ba8KIbxA">

Thank you huangbaichao,
That was helpful. It's very hard to get useful informations about the ComfyUI API

**Comment by greenzorro** [comment]
> Got HTTP Error 400: Bad Request. Is the request data properly set?

I also encountered the same problem. There may be many reasons for this problem. Later I found that my situation was caused by the model storage path problem. Due to my own special needs, I created a new subdirectory in checkpoints to store the checkpoint model. Such a directory structure can be manually generated directly in comfyui, but a 400 error will be reported through API calls. I put the model back into the checkpoints directory and it is normal.
